File: c/v2.0.2022/prime.c
#include <stdio.h>
#include <time.h>
#include <math.h>
#include <string.h>

int main()
{
    int last = 10000;
    int found = 4;   // we already know 2, 3, 5, 7
    const int arraylength = (int)(last / log(last));
    int primes[1000000] = {2, 3, 5, 7};
    clock_t start, end;
    double cpu_time_used;
    printf("Calculating prime numbers until %d\n", last);
    start = clock();
    for (int number = 11; number < last; number += 2)
    {
        int prime = 1;
        for (int divider = 3; divider < (int)(sqrt(number)) + 1; divider += 2)
        {
            if (number % divider == 0)
            {
                prime = 0;
                break;
            }
        }
        if (prime ==  1)
        {
            // primes[found] = number;
            found += 1;
        }
    }
    end = clock();
    cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
    // for (int i = 0; i < found - 1; i++)
    // {
    //     printf("%d, ",primes[i]);
    // }
    printf("\nFound %d prime numbers.\n", found);
    printf("This took %f seconds.",cpu_time_used);
}

File: python/v2.0.2022/prime.py
# prime v2.0.2022 with modulo operator 
import math, time
last = 1000000         # 1 million is a good comparison value
found = 4              # we start from 11, know 2, 3, 5, 7
print(f"Prime numbers to {last} with algorithm v2.0.2022")
start = time.perf_counter_ns()
for number in range(11, last, 2):
    prime = True
    for divider in range(3, int(math.sqrt(number))+1, 2):
        if number % divider == 0:
            prime = False
            break
    if prime:
        found += 1
end = time.perf_counter_ns()
print(f"This took: {(end - start)/1000} microseconds.")
print(f"I found {found} prime numbers. Should be 78498.")
